asserts/system_user.go

```diff
@@ -216,6 +246,30 @@ func checkHashedPassword(headers map[string]interface{}, name string) (string, e
 	return pw, nil
 }
 
+func parseSystemUserExpiration(headers map[string]interface{}, until time.Time) (string, error) {
+	value, ok := headers["user-valid-for"]
+	if !ok {
+		return "", nil
+	}
+
+	str, ok := value.(string)
+	if !ok {
+		return "", fmt.Errorf("cannot parse 'user-valid-for': not a string")
+	}
```

Comment:
this could and shuld use checkOptionalString

Reply:
Done, thanks for pointing this out!